adr/keycloak-adr.md

```diff
@@ -0,0 +1,139 @@
+# ADR: Enhanced Integration of Keycloak with GitLab for Authentication and User Management
```

Comment:
Hey Naveen, in case you haven't seen it, take a look [here](https://github.com/defenseunicorns/uds-software-factory/blob/main/CONTRIBUTING.md#documentation) for more info on the ADR file name format and the title. 

Generally the file name should start with a sequential number that's also referenced in the title. That gives a way to have a later ADR supersede or amend a previous one based on the ADR number.

The link I included contains links to an article that explain it, and you can see a good example of how it looks in the zarf repo where it is more built out. https://github.com/defenseunicorns/zarf/tree/main/adr

Reply:
Thanks, @ericwyles! I have updated the file name. 👍